vscode/src/editor/vscode-editor.ts

```diff
@@ -157,6 +157,56 @@ export class VSCodeEditor implements Editor<InlineController, FixupController, C
         return null
     }
 
+    /**
+     * This function retrieves a "smart" selection from the active editor for a "fixup" recipe
+     * The idea of a "smart" selection is to look at both the start and end positions of the current selection,
+     * and attempt to expand those positions to encompass more meaningful chunks of code, such as folding regions.
+     *
+     * The function does the following:
+     * 1. Checks if there's an active editor.
+     * 2. Determines the current selection's start and end line numbers.
+     * 3. For both the start and end line numbers, it tries to find a more encompassing folding range.
+     *    - If a folding range is found, it uses that. Otherwise, it defaults to the original selection.
+     * 4. The function then creates a new selection that starts at the beginning of the expanded start range
+     *    and ends at the end of the expanded end range.
+     * 5. Returns this combined range as the final "smart" selection.
```

Comment:
These comments make it sound more complicated than it is.

```suggestion
     * 2. If the selection starts in a folding range, moves the selection start back to the start of that folding range.
     * 3. If the selection ends in a folding range, moves the selection end forward to the end of that folding range.
```

Reply:
Good catch, I changed this.